File: Algorithms/01-simple-array-sum/01.cpp
#include <iostream>
#include <vector>

using namespace std;

int simpleArraySum(vector <int> ar) {
    int result = 0;
    for (int x = 0; x < ar.size(); x++)
        result += ar[x];
    return result;
}

int main() {
    int n;
    cin >> n;
    vector<int> ar(n);
    for(int ar_i = 0; ar_i < n; ar_i++){
       cin >> ar[ar_i];
    }
    int result = simpleArraySum(ar);
    cout << result << endl;
    return 0;
}

File: Algorithms/01-simple-array-sum/01.rb
#!/usr/local/bin/ruby -w

def simple_array_sum(ar)
    sum = 0
    ar.each do |number|
        sum += number
    end
    sum
end

n = gets.strip.to_i
ar = gets.strip
ar = ar.split(" ").map(&:to_i)
result = simpleArraySum(ar)
puts result
